src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import {
+  ComponentBaseProps,
+  generateRandomString,
+  pxToRem,
+} from '../../shared';
+import styled from 'styled-components';
+
+export interface SidebarProps extends ComponentBaseProps<HTMLDivElement> {
+  /**
+   * Header icon
+   */
+  icon?: React.ReactNode;
+  /**
+   * Sidebar Slot
+   */
+  slot?: React.ReactNode;
+  /**
+   * Show overlay
+   */
+  overlay?: boolean;
+}
+/**
+ * Sidebar and overlay
+ */
+const SidebarWrapper = styled.div<SidebarProps>`
+  height: 100%;
+  display: flex;
+  flex-direction: row;
+`;
+/**
+ * Sidebar overlay
+ */
+const SidebarOverlay = styled.div<SidebarProps>`
+  height: 100%;
+  width: 100%;
+  background-color: black;
+`;
+/**
+ * Wrapper for sidebar
+ */
+const SidebarBody = styled.div<SidebarProps>`
```

Comment:
Tämä tarvisi dropshadow-tyylin, joka on figmassa. Mutta se pitänee laittaa noihin teema-tiedostoihin jotenkin, että tumman ja vaalean teeman dropshadow tulee oikein. Eli varmaan jotenkin näin kumpaankin teemaan
`colors: {...},
styles: {
  dropshadow: 'box-shadow: 0px 8px 25px 0px rgba(0, 0, 0, 0.15);'
  }`

Reply:
dark: box-shadow: 0px 10px 25px 0px rgba(0, 0, 0, 0.25);